### PR TITLE
Improved Assessment Configurability in Ground Control

### PR DESCRIPTION
**Pull request by DesSnowy**
This pull request supports user to configure assessment setting per assessment. This allows better control of each individual assessments and which features they have.

**Changes to be made:**

- [X] Added "hasTokenCounter" and "hasVotingFeatures" to assessments
- [X] Added "hasVotingFeatures" to assessmentConfig
- [X]  "hasTokenCounter" and "hasVotingFeatures" will be initialised to the respective value in assessmentConfig when the assessment is uploaded
- [X] Changed output to show settings of individual assessments
- [X] Updated test case to match

### CODE REVIEW

**Comment by DesSnowy** [comment]
Corresponding frontend PR: https://github.com/source-academy/frontend/pull/2863

File: lib/cadet_web/admin_controllers/admin_assessments_controller.ex [hunk 1]
```diff
@@ -98,6 +100,15 @@ defmodule CadetWeb.AdminAssessmentsController do
...
 
+    updated_assessment =
+      if is_nil(has_token_counter) do
+        updated_assessment
+      else
+        updated_assessment
+        |> Map.put(:has_token_counter, has_token_counter)
+        |> Map.put(:has_voting_features, has_voting_features)
+      end
+
```
Comment (RichDom2185) [inline lib/cadet_web/admin_controllers/admin_assessments_controller.ex:111]: I feel like this should be two separate "if" checks, one for each param&hellip;. Thoughts?

**Review by RichDom2185 — COMMENTED** [review]
Just the one comment, thanks!

File: lib/cadet_web/admin_controllers/admin_assessments_controller.ex (hunk 1 above)
Comment (DesSnowy) [reply lib/cadet_web/admin_controllers/admin_assessments_controller.ex]: Yea, I think that's better. I have changed it to be separated now. Let me know if there's any other issues!

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**